### PR TITLE
SW-6282 Added isAdHoc flag and ObservationType to observation and monitoring plot tables

### PR DESCRIPTION
**Pull request by tommylau523**

### CODE REVIEW

**Comment by tommylau523** [comment]
* **#2681** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2681?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2681?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/main/resources/db/migration/0300/V326__AdHocPlots.sql [hunk 1]
```diff
@@ -0,0 +1,27 @@
+CREATE TABLE tracking.observation_types (
+    id INTEGER PRIMARY KEY,
+    name TEXT NOT NULL UNIQUE
+);
+
+INSERT INTO tracking.observation_types (id, name)
+VALUES (1, 'Monitoring'),
+       (2, 'Biomass Measurements')
+ON CONFLICT (id) DO UPDATE SET name = excluded.name;
```
Comment (sgrimm) [inline src/main/resources/db/migration/0300/V326__AdHocPlots.sql:9]: This should go in `R__TypeCodes.sql` too.

**Review by sgrimm — APPROVED** [review]
Looks great! Just one tiny thing to add before merging.

**Comment by sgrimm** [comment]
We should add these new fields to the search API too, but that can be a separate PR.

**Comment by tommylau523** [comment]
Added!

**Review by sgrimm — APPROVED** [review]

**Pull request merged**